pkg/provider/k8s/k8s.go

```diff
@@ -16,62 +17,104 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
-	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/util/retry"
 
 	"strings"
 
 	"github.com/prometheus/prombench/pkg/provider"
+
+	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
-// K8s is the main provider struct.
+// Resource holds the resource objects obtained after parsing deployment files.
+type Resource struct {
+	FileName string
+	Objects  []runtime.Object
+}
+
+// K8s holds the fields used to generate API reqeust from within a cluster.
 type K8s struct {
 	clt *kubernetes.Clientset
+	// DeploymentFiles files provided from the cli.
+	DeploymentFiles []string
+	// Vaiables to subtitude in the DeploymentFiles.
+	// These are also used when the command requires some variables that are not provided by the deployment file.
+	DeploymentVars map[string]string
+	// DeploymentFile content after substituting the variables filename is used as the map key.
+	K8sResource []Resource
```

Comment:
update comment
and can be just as `Resources` to reduce the stutter.

and maybe this shouldn't be exported and use a getter as: K8s.Deployments()